solvis/solution/typing.py

```diff
@@ -32,42 +32,52 @@ class InversionSolutionFileProtocol(Protocol):
     @property
     def fault_regime(self) -> str:
         """solution requires a fault regime"""
+        raise NotImplementedError()
 
     @property
     def average_slips(self) -> gpd.GeoDataFrame:
         """the average slips for each rupture."""
+        raise NotImplementedError()
 
     @property
     def section_target_slip_rates(self) -> gpd.GeoDataFrame:
         """the inversion target slip rates for each rupture."""
+        raise NotImplementedError()
 
     @property
     def indices(self) -> gpd.GeoDataFrame:
         """the fault sections involved in each rupture."""
+        raise NotImplementedError()
 
     @property
     def archive_path(self) -> Optional[Path]:
         """the path to the archive file"""
+        raise NotImplementedError()
 
     @property
     def archive(self) -> Optional[zipfile.ZipFile]:
         """the archive instance"""
+        raise NotImplementedError()
 
 
 class AggregateSolutionFileProtocol(Protocol):
     @property
     def fast_indices(self) -> gpd.GeoDataFrame:
-        pass
+        """enable fast indices"""
```

Comment:
`raise NotImplementedError()`

Reply:
@chrisdicaprio yep, also in next PR. thanks